frontend/src/__tests__/cypress/cypress/tests/mocked/projects/workbench.cy.ts

```diff
@@ -399,10 +400,72 @@ describe('Workbench page', () => {
     });
 
     cy.get('@editWorkbench.all').then((interceptions) => {
-      expect(interceptions).to.have.length(2); // 1 dry run request and 1 actaul request
+      expect(interceptions).to.have.length(2); // 1 dry run request and 1 actual request
     });
   });
 
+  it('Handle deleted notebook sizes in workbenches table', () => {
+    initIntercepts({});
+    cy.interceptOdh(
+      'GET /api/config',
+      mockDashboardConfig({
+        notebookSizes: [
+          {
+            name: 'Medium',
+            resources: {
+              limits: {
+                cpu: '6',
+                memory: '24Gi',
+              },
+              requests: {
+                cpu: '3',
+                memory: '24Gi',
+              },
+            },
+          },
+          {
+            name: 'Large',
+            resources: {
+              limits: {
+                cpu: '14',
+                memory: '56Gi',
+              },
+              requests: {
+                cpu: '7',
+                memory: '56Gi',
+              },
+            },
+          },
+          {
+            name: 'X Large',
+            resources: {
+              limits: {
+                cpu: '30',
+                memory: '120Gi',
+              },
+              requests: {
+                cpu: '15',
+                memory: '120Gi',
+              },
+            },
+          },
+        ],
+      }),
+    );
+    workbenchPage.visit('test-project');
+    const notebookRow = workbenchPage.getNotebookRow('Test Notebook');
+    notebookRow.shouldHaveNotebookImageName('Test Image');
+    notebookRow.shouldHaveContainerSize('Custom');
+    notebookRow.findKebabAction('Edit workbench').click();
+    editSpawnerPage.shouldHaveContainerSizeInput('Custom');
+    workbenchPage.visit('test-project');
```

Comment:
Does this cause a page refresh in the Cypress UI?

Reply:
yes

Reply:
@christianvogt How do you redirect without a refresh?

Reply:
Looks to me like you just want to go back to the previous page. In which case you can manipulate the history using `cy.go('back')`. There's no way to go directly to a URL without a page refresh AFAIK. Therefore another alternative is to utilize the app navigation and manually navigate.

Reply:
This test is failing without a page refresh for me: `createSpawnerPage.findContainerSizeInput('Keep custom size').should('not.exist');`

Reply:
Wait, why is it failing? Did you find a bug that you're trying to just ignore or is this a cypress-ism and some state is weirdly set because of it? Did you debug the issue @manaswinidas ? Have you ran the test manually to see if the issue is there too?

Reply:
Fixed after changes following [this comment](https://github.com/opendatahub-io/odh-dashboard/pull/2944#discussion_r1679436931)